src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp

```diff
@@ -257,18 +341,35 @@ template<UnsignedInt dimensions> Containers::Optional<Containers::Array<char>> c
     /* If these are enabled, the library reads BMPs/JPGs/PNGs/TGAs from the
        filesystem and then writes basis files there also. DO NOT WANT. */
     params.m_read_source_images = false;
+    #if BASISU_LIB_VERSION >= 150
+    params.m_write_output_basis_or_ktx2_files = false;
+    #else
     params.m_write_output_basis_files = false;
+    #endif
+
     /* One image per slice. The base mip is in m_source_images, mip 1 and
        higher go into m_source_mipmap_images. */
     const UnsignedInt numImages = Vector3i::pad(baseSize, 1).z();
-    params.m_source_images.resize(numImages);
-    if(numMipmaps > 1) {
-        params.m_source_mipmap_images.resize(numImages);
-        for(auto& slice: params.m_source_mipmap_images)
-            slice.resize(numMipmaps - 1);
+    if(isHdr) {
+        #if BASISU_LIB_VERSION >= 150
+        params.m_source_images_hdr.resize(numImages);
+        if(numMipmaps > 1) {
+            params.m_source_mipmap_images_hdr.resize(numImages);
+            for(auto& slice: params.m_source_mipmap_images_hdr)
+                slice.resize(numMipmaps - 1);
+        }
+        #else
+        CORRADE_INTERNAL_ASSERT_UNREACHABLE();
```

Comment:
The problem with the unreachable assert is that it will fail only at runtime if some refactoring goes wrong and `isHdr` becomes set. I think a better way would be to wrap also `if(isHdr)`, the `bool isHdr` and everything related in `#if BASISU_LIB_VERSION >= 150`. Then a lot more of the correctness is checked at compile time. Same with the unreachable assert below.

Related -- to avoid accidental breakages in existing projects that bundle older Basis, at least one CI job should stay at the old version. Can you update the configs in `package/ci` to make some stay at older and some at newer? I think e.g. the MSVC 2015/2017 could be at 1.15, 2019 and MinGW at 1.16, 2022 and clang-cl at 1.50 and then linux + mac at 1.16 / 1.50, because those are so fragmented that we want to verify that OpenCL works on all.

(I can do that myself too, but there aren't any prebuilt packages so it should be doable from your end as well).